src/omt/omt_optimizer.cpp

```diff
@@ -17,20 +17,22 @@
 
 #include "omt/bitvector_optimizer.h"
 #include "omt/integer_optimizer.h"
-#include "options/smt_options.h"
-#include "smt/smt_engine.h"
-#include "theory/quantifiers/quantifiers_attributes.h"
-#include "theory/smt_engine_subsolver.h"
 
 using namespace cvc5::theory;
 using namespace cvc5::smt;
 namespace cvc5::omt {
 
-std::unique_ptr<OMTOptimizer> OMTOptimizer::getOptimizerForNode(TNode targetNode,
-                                                                bool isSigned)
+bool OMTOptimizer::nodeSupportsOptimization(TNode node)
+{
+  TypeNode type = node.getType();
+  return (type.isInteger() || type.isBitVector());
```

Comment:
I see some code that treats also Real and FP, so it is not clear why they are not considered in this line. Please make this clearer in the comments in the code.

Reply:
they are not supported. the other code adds Real and FP but they really are not supported for optimization.